### PR TITLE
Queue implementation

### PR DESCRIPTION
**Pull request by HarryWillis0**
any issues?

### CODE REVIEW

File: queue/ArrayQueue.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    /**
+     * dequeue()
+     * removes element at the front of the queue
+     * @return item removed
+     * @throws exception if queue empty
+     */
+    public T dequeue() throws Exception{
+        if(this.back  == 0){
```
Comment (SamuelWillis) [inline queue/ArrayQueue.java]: Two spaces between `this.back` and `==`

File: queue/ArrayQueue.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+     */
+    public ArrayQueue(){
+        this(5);
+    }
+
+    /**
+     * Build ArrayQueue with specified size
+     */
+    @SuppressWarnings("unchecked")
+    public ArrayQueue(int size){
```
Comment (SamuelWillis) [inline queue/ArrayQueue.java]: This is a stylistic thing but generally, it's nice to leave a space before the `{`

I like that you've been consistent with your spacing so if you prefer it this way then leave. Key is to be consistent.

File: queue/ArrayQueue.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
+import java.util.Arrays;
+/**
```
Comment (SamuelWillis) [inline queue/ArrayQueue.java]: Space between comment and import.

File: queue/ArrayQueue.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+ * An array implementation of a queue
+ * 
+ * @author Harry Willis
+ */
+
```
Comment (SamuelWillis) [inline queue/ArrayQueue.java]: No space here.

File: queue/ArrayQueue.java [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        this.queue = (T[]) new Object[size];
+    }
+    /**
+     * enqueue(T item)
+     * add an element to the back of the queue, extending capacity if necessary
+     * @param item to be added
+     * @return item added
+     */
+    public T enqueue(T item){
+        if(this.back + 1 == this.size){
```
Comment (SamuelWillis) [inline queue/ArrayQueue.java]: Similar to my stylistic comment above, generally ifs and whiles and fors etc will be written with spaces between their components for legibility.

`if (this.back + 1 == this.size) {`

Again, you've been consistent throughout the code and that's key and if you prefer how it is now, feel free to leave it.

File: queue/ArrayQueue.java [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+            temp[i] = this.queue[i];
+        }
+        
+        this.size *= 2;
+        
+        this.queue = temp;
+        
+        return this.queue;
+    }
+
```
Comment (SamuelWillis) [inline queue/ArrayQueue.java]: extra space here

File: stack/ArrayStack.java [hunk 7]
```diff
@@ -5,16 +5,14 @@
  * 
  * @author Harry Willis
  */
-
-public class ArrayStack<T extends Comparable <T>> implements Stack<T> {
+public class ArrayStack<T extends Object> implements Stack<T> {
```
Comment (SamuelWillis) [inline stack/ArrayStack.java]: Why does T extend Object here? This new to me :)

File: stack/ArrayStack.java [hunk 8]
```diff
@@ -35,52 +33,125 @@ public ArrayStack(int size){
...
+        return item;
     } 
+
+    /**
+     * Search for item in the stack
+     * @return the distance from the top of the stack of the closest to the top occurence
+     *          -1 if item does not occur on stack
+     * @throws exception if stack is empty 
+     */
     public int search(T item) throws Exception {
```
Comment (SamuelWillis) [inline stack/ArrayStack.java]: What does this function do? I haven't seen `search` as a stack function before.

File: queue/ArrayQueue.java (hunk 2 above)
Comment (HarryWillis0) [reply queue/ArrayQueue.java]: It's just habit. I like with the space more.

File: stack/ArrayStack.java (hunk 8 above)
Comment (HarryWillis0) [reply stack/ArrayStack.java]: I followed - https://docs.oracle.com/javase/7/docs/api/java/util/Stack.html

File: stack/ArrayStack.java (hunk 7 above)
Comment (HarryWillis0) [reply stack/ArrayStack.java]: This is for the search function. It ensures that whatever type is being used for the stack is a subclass of Object and thus has an equals function, which I use in search().

**Pull request merged**